modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala

```diff
@@ -0,0 +1,776 @@
+      offset = None,
+      weight = None
+    )
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.DecayFunction]] with `decayFunctionType` of
+   * [[zio.elasticsearch.query.DecayFunctionType.Exp]] using the specified parameters.
+   * [[zio.elasticsearch.query.DecayFunction]] is used to score a document with a function that decays depending on the
+   * distance of a numeric field value of the document from a user given origin. This is similar to a
+   * [[zio.elasticsearch.query.RangeQuery]], but with smooth edges instead of boxes.
+   *
+   * @param field
+   *   the field for which query is specified for
+   * @param origin
+   *   the point of origin used for calculating distance
+   * @param scale
+   *   defines the distance from origin + offset at which the computed score will equal `decay`
+   * @return
+   *   an instance of [[zio.elasticsearch.query.DecayFunction]] that represents the function score query to be
+   *   performed.
+   */
+  def expDecayFunction(field: String, origin: String, scale: String): DecayFunction[Any] =
+    DecayFunction[Any](
+      field = field,
+      decayFunctionType = Exp,
+      origin = origin,
+      scale = scale,
+      decay = None,
+      filter = None,
+      multiValueMode = None,
+      offset = None,
+      weight = None
+    )
+
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.query.FieldValueFactor]] using the specified parameters.
+   * [[zio.elasticsearch.query.FieldValueFactor]] function allows you to use a field from a document to influence the
+   * score. It is similar to using the script_score function, however, it avoids the overhead of scripting. If used on a
+   * multi-valued field, only the first value of the field is used in calculations.
+   *
+   * @param field
+   *   the type-safe field to be extracted from the document
+   * @return
+   *   an instance of [[zio.elasticsearch.query.FieldValueFactor]] that represents the function score query to be
+   *   performed.
+   */
+  def fieldValueFactor[S](field: Field[S, _]): FieldValueFactor[S] =
+    FieldValueFactor[S](
+      field = field.toString,
+      factor = None,
+      filter = None,
+      modifier = None,
+      missing = None,
+      weight = None
+    )
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.FieldValueFactor]] using the specified parameters.
+   * [[zio.elasticsearch.query.FieldValueFactor]] function allows you to use a field from a document to influence the
+   * score. It is similar to using the [[zio.elasticsearch.query.ScriptScoreFunction]], however, it avoids the overhead
+   * of scripting. If used on a multi-valued field, only the first value of the field is used in calculations.
+   *
+   * @param field
+   *   the field to be extracted from the document
+   * @return
+   *   an instance of [[zio.elasticsearch.query.FieldValueFactor]] that represents the
+   *   [[zio.elasticsearch.query.FunctionScoreQuery]] to be performed.
```

Comment:
In some places you use reference and in some places you just use `function score query` for same thing (in @return sentence).